doc/sphinx/dev_git_intro.rst

```diff
@@ -52,46 +52,211 @@ Clone a local repository onto your machine
+::
+  git add file1
+  git add file2
+  ...
+  git rm file3
+4. Next, continue the rebase
+::
+  git rebase --continue
+
+The editor will open with the modified commit history. Simply save the changes and close the editor (``ESC+SHIFT+wq``), and the rebase will continue. If the rebase stops with errors, repeat the merge conflict resolution process, add/remove the files to staging area, type ``git rebase --continue``, and proceed.
+
+If you have not updated your branch in a long time, you'll likely find that you have to keep fixing the same conflicts over and over again (every time your commits collide with the commits on the main branch). This is why we strongly advise POD developers to pull updates into their forksand rebase their branches onto the develop branch frequently.
+
+Note that if you want to stop the rebase at any time and revert to the original state of your branch, type
+::
+  git rebase --abort
+
+5. Once the rebase has completed, push your changes to the remote copy of your branch
+::
+  git push origin feature/<my_feature_name> --force
+The ``--force`` option is necessary because rebasing modified the commit history.
+
+6. Now that your branch is up-to-date, write your code!
 
-- If you've added new files, ``git add --all`` before commit the changes.
+Pushing to your remote POD development branch on your fork
+----------------------------------------------------------
+When you are ready to push your updates to the remote branch on your fork
 
-- Commit changes with ``git commit -a``. This creates a snapshot of the code into the history in your local repo.
+1. type ``git status`` to list the file(s) that have been updated
 
+2. type ``git add <file>`` to add individual files, or ``git add --all`` to add all files, that have been updated to the staging area
+
+3. Commit the changes with ``git commit -m <your commit message>``. You can also type ``git commit`` to launch an editor in the terminal where you can enter your message.
+
+If you use the editor or BASH shell, you can easily break up your message over multiple lines for better readability.
+
+4. Push the updates to your fork: ``git push -u origin feature/<my_feature_name>`` (The ``-u`` flag is for creating a new branch remotely and only needs to be used the first time.)
+
+Pull Requests
+-------------
+A Pull Request (PR) is your proposal to the maintainers to incorporate your feature into NOAA's repo. When your feature is ready, submit a PR by going to the GitHub page of your fork and clicking on **Pull request** to the right of the branch description. Make sure you are submitting the PR to NOAA-GFDL/develop. Enter a brief description for the PR, and check the boxes in the to-do list for the completed tasks. If you are still working on your POD, but want to test it with the CI, you can select the *Create Draft Pull Request* option from the dropdown menu by clicking the green button with the arrow to the right of the **Create Pull Request Button**.
```

Comment:
What is the CI?

Since I haven't seen anything about merging our feature branch onto (our) origin/develop, I assume your advice is to do the pull request from the feature branch? If so, perhaps add that in "to the right of your feature branch description"

Reply:
You should submit your feature branch directly as the PR. It is best practice to leave the main and develop branches on your fork alone unless you are syncing them with the NOAA-GFDL fork.

The CI is currently the lightweight build test that runs on TravisCI (when there enough credits), and the LGTM security analysis/code linter.

Reply:
Nice to know about leaving the main & develop branches of my fork alone; I had thought I should update my features to my develop branch (analogous to the upstream process). Might be nice to suggest this upfront (if I didn't just miss it).